flood_api/routers/flood.py

```diff
@@ -1,21 +1,125 @@
-from fastapi import APIRouter
+from fastapi import APIRouter, HTTPException
+from fastapi.responses import JSONResponse
 
-from flood_api.dependencies.flooddata import DetailedDataDep
-from flood_api.dependencies.queryparams import CoordinatesDep, DateRangeDep
+from flood_api.dependencies.flooddata import (DetailedDataDep,
+                                              SummaryDataDep,
+                                              ThresholdDataDep)
+from flood_api.dependencies.queryparams import (CoordinatesDep, 
+                                                DateRangeDep,
+                                                IncludeNeighborsDep)
+from flood_api.utils.geospatial_operations import get_data_for_point
+from flood_api.utils.json_utilities import (dataframe_to_geojson, 
+                                            custom_date_handler)
+from flood_api.utils.validation_helpers import point_within_roi
 
 router = APIRouter(tags=["flood"])
 
+@router.get("/summary", response_class=JSONResponse)
+async def summary(
+    gdf: SummaryDataDep,
+    coordinates: CoordinatesDep,
+    include_neighbors: IncludeNeighborsDep,
+):
+    longitude, latitude = coordinates
+
+    # Check if the queried point is within the ROI
+    if not point_within_roi(latitude=latitude, longitude=longitude):
+        raise HTTPException(status_code=400, detail="Queried coordinates are outside the region of interest")
```

Comment:
Hmm...Is this the best approach for handling queries where we have no data?
Can we simply not return 200 with no data, or will that perhaps be misleading.

Reply:
It might be useful to indicate that we don't currently support queries to some regions. I am afraid that users will interpret no flood data as us saying that there is no flood risk in that area, when it's just because we're not processing data for that area yet

Reply:
Yes, I agree with that, but perhaps a 404 is a better http-status? We can also in the details indicate why the 404.

Reply:
As long as an explanation is provided, I'm fine with 404

Reply:
The response for queried points that lie outside the ROI would then be identical to points within the ROI that don't have a food forecast. I think this could be a bit misleading as our API is not intended to serve data for points that outside the defined ROI.